### PR TITLE
Update engine configuration file

### LINKED ISSUE #46: Update the  Engine configuration file documentation
Document the updates to the config files.

Example files can be found here:
- Manager: https://github.com/ConsenSys/sumo/tree/main/manager/config
- Engine: https://github.com/ConsenSys/sumo/tree/main/engine/config

Here's a list of commits made in the Engine directory (which may be able to provide some context)
https://github.com/ConsenSys/sumo/commits/main/engine/config

### PR DESCRIPTION
**Pull request by rolandtyler**
## Pull Request Description
Update engine configuration options
## Fixed Issue(s)

Fixes #46 

## Preview
https://pegasys-sumo--57.com.readthedocs.build/en/57/Reference/Configuration-File/#engine-configuration

### CODE REVIEW

**Review by citizen-stig — APPROVED** [review]

File: docs/Reference/Configuration-File.md [hunk 1]
```diff
@@ -289,19 +284,16 @@ Gas limit for a call to `submitTransactions` in the rollup smart contract. The d
...
-#### `lookahead`
+#### `lookahead_blocks`
 
 Maximum number of blocks to fetch in a single query when the operator is catching up.
 
-#### `nb_of_blocks_to_finalization`
+#### `blocks_to_finalization`
 
 Number of successors a block requires before being considered final.
+Set it to `1` for a private deployment, `3` or more for mainnet.
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:294]:
```suggestion
Set to `1` for a private deployment, or `3` or more for Mainnet.
```

File: docs/Reference/Configuration-File.md [hunk 2]
```diff
@@ -349,12 +341,16 @@ This section contains the [Kafka](https://kafka.apache.org/) settings:
...
-Topic used by Kafka. The default is `sumo-state-updates`.
+Topic used by Kafka. The default is `operator1_state_updates`.
 
 #### `brokers`
 
 Address of the [Kafka brokers]. The default is `localhost:9092`.
 
+#### `invalid_operations_topic`
+
+Topic where information about invalid operations are sent. The default is `operator1_invalid_operations`
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:352]:
```suggestion
Topic where information about invalid operations are sent. The default is `operator1_invalid_operations`.
```

File: docs/Reference/Configuration-File.md [hunk 3]
```diff
@@ -436,34 +432,68 @@ Name of the [store](https://docs.quorum-key-manager.consensys.net/en/stable/Conc
...
 
 The arity of the rollup. The default is `2`.
 
-#### `capacity`
+#### `balance_merkle_tree_depth`
 
-Targeted capacity of the rollup to ensure memory is immediately allocated correctly. Can be changed
-after the rollup's creation.
+Depth of the account's balance tree.
+The rollup's maximum capacity (in number of tokens) is 2^balance_merkle_tree_depth.
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:468]:
```suggestion
The rollup's maximum capacity (in number of tokens) is $2^\text{balance_merkle_tree_depth}$.
```

File: docs/Reference/Configuration-File.md [hunk 4]
```diff
@@ -436,34 +432,68 @@ Name of the [store](https://docs.quorum-key-manager.consensys.net/en/stable/Conc
...
 
-#### `depth`
+#### `money_order_batch_merkle_tree_depth`
 
-The depth of the rollup's Merkle tree. The rollup's maximum capacity is $2^\text{depth}$.
-The parameter cannot be changed after the rollup's creation.
+Depth of the money order batch.
+The maximum number of money orders that can be created/redeemed in a single batch is 2^money_order_batch_merkle_tree_depth
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:473]:
```suggestion
The maximum number of money orders that can be created/redeemed in a single batch is $2^\text{money_order_batch_merkle_tree_depth}$.
```

File: docs/Reference/Configuration-File.md [hunk 5]
```diff
@@ -436,34 +432,68 @@ Name of the [store](https://docs.quorum-key-manager.consensys.net/en/stable/Conc
...
+Applicable only for [partially anonymous rollups](../Concepts/Rollups/Partially-Anonymous-Rollups.md).
+
+#### `money_order_initial_capacity`
+
+Capacity pre-allocated for money order history call tree
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:478]:
```suggestion
Capacity pre-allocated for money order history call tree.
```

File: docs/Reference/Configuration-File.md [hunk 6]
```diff
@@ -436,34 +432,68 @@ Name of the [store](https://docs.quorum-key-manager.consensys.net/en/stable/Conc
...
+Applicable only for [partially anonymous rollups](../Concepts/Rollups/Partially-Anonymous-Rollups.md).
+
+#### `money_order_merkle_tree_depth`
+
+Depth of money orders history call tree, must be greater than 1 and less than or equal to 32.
+The maximum number of money order batches that can be created is 2^money_order_merkle_tree_depth.
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:484]:
```suggestion
The maximum number of money order batches that can be created is $2^\text{money_order_merkle_tree_depth}$.
```

File: docs/Reference/Configuration-File.md [hunk 7]
```diff
@@ -436,34 +432,68 @@ Name of the [store](https://docs.quorum-key-manager.consensys.net/en/stable/Conc
...
+Applicable only for [partially anonymous rollups](../Concepts/Rollups/Partially-Anonymous-Rollups.md).
 
-#### `thread_count`
+#### `thread_pool_thread_count`
 
-Number of threads to use when recalculating the hash tree during startup. Set this to the number of
-cores available.
+Number of threads to use for signatures validation and root hash calculation.
```
Comment (alexandratran) [inline docs/Reference/Configuration-File.md:489]: signature validation?

**Review by bgravenorst — APPROVED** [review]
lgtm

**Pull request merged**